src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/loptimizer/heuristics/utils/Join_Analyzer.java

```diff
@@ -0,0 +1,75 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.loptimizer.heuristics.utils;
+
+import org.apache.jena.graph.Node;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Join_Analyzer {
+
+    // Consider the subject-subject and object-object joins as star joins
+    public static double getNumOfStarJoins( final List<Node> vars_s, final List<Node> vars_o ){
+        final int jsubs = countDuplicates(vars_s);
+        final int jobjs = countDuplicates(vars_o);
+        return ( jsubs + jobjs );
+    }
+
+    // Consider the object-subject and subject-object joins as chain joins
```

Comment:
What does "consider" mean? What is the purpose of this function/what does it do?